app/tests/test_domain_commands.py

```diff
@@ -31,3 +43,87 @@ def test_resultCommand_invokeUponSamplePool_resultIsInPool():
     given_command.invoke(given_pool)
 
     assert given_pool.results[given_id] == given_result
+
+
+def test_registerCommand_invokeNoSubproblemBeingComputed_returnsEmptyList():
+    given_id = TestId(42)
+    given_pool = TestPool()
+    given_command = RegisterCommand(given_id)
+
+    register_invoke_output = given_command.invoke(given_pool)
+
+    assert len(register_invoke_output) == 0
+
+
+def test_registerCommand_invokeWithDifferentSubproblemInProgress_returnsEmptyList():
+    given_id = TestId(42)
+    current_id = TestId(16)
+    given_pool = TestPool()
+    given_command = RegisterCommand(given_id)
+    given_pool.in_progress_pool.add(current_id)
+    given_pool.current_subproblem_id = current_id
+
+    register_invoke_output = given_command.invoke(given_pool)
+
+    assert len(register_invoke_output) == 0
+
+
+def test_registerCommand_invokeSenderHasGreaterPriority_returnsEmptyList():
+    given_id = TestId(42)
+    given_pool = TestPool()
+    given_command = RegisterCommand(given_id)
+    given_pool.in_progress_pool.add(given_id)
+    given_pool.current_subproblem_id = given_id
+
+    sender_address = ConnectionSettingsFactory.higher_priority_address()
+    local_address = ConnectionSettingsFactory.lower_priority_address()
+    given_command.context.initialize(sender_address, local_address)
+
+    register_invoke_output = given_command.invoke(given_pool)
+
+    assert len(register_invoke_output) == 0
+    assert given_pool.current_subproblem_id == None
+
+
+def test_registerCommand_invokeSubproblemInProgressOnDifferentAgent_returnsEmptyList():
+    given_id = TestId(42)
+    given_pool = TestPool()
+    given_command = RegisterCommand(given_id)
+    given_pool.in_progress_pool.add(given_id)
+
+    register_invoke_output = given_command.invoke(given_pool)
+
+    assert len(register_invoke_output) == 0
+
+
+def test_registerCommand_invokeWithSubproblemAlreadyInResults_returnsResult():
+    given_id = TestId(42)
+    given_result = TestResult(16)
+    given_pool = TestPool()
+    given_command = RegisterCommand(given_id)
+    given_pool.results[given_id] = given_result
+
+    register_invoke_output = given_command.invoke(given_pool)
+    
+    assert len(register_invoke_output) == 1
+    assert isinstance(register_invoke_output[0], BaseResultCommand)
```

Comment:
Use get_identifier(). Checking for BaseXCommand type is fiendishly implementation-specific.